src/Nri/Ui/TextInput/V6.elm

```diff
@@ -0,0 +1,316 @@
+-}
+onBlur : msg -> Attribute value msg
+onBlur msg =
+    OnBlurAttribute msg
+
+
+{-| Sets the `autofocus` attribute of the resulting HTML input.
+-}
+autofocus : Attribute value msg
+autofocus =
+    AutofocusAttribute True
+
+
+{-| This is private. The public API only exposes `Attribute`.
+-}
+type alias Config msg =
+    { inputStyle : InputStyles.Theme
+    , isInError : Bool
+    , hideLabel : Bool
+    , placeholder : Maybe String
+    , onBlur : Maybe msg
+    , autofocus : Bool
+    }
+
+
+emptyConfig : Config msg
+emptyConfig =
+    { inputStyle = InputStyles.Standard
+    , isInError = False
+    , hideLabel = False
+    , placeholder = Nothing
+    , onBlur = Nothing
+    , autofocus = False
+    }
+
+
+updateConfig : Attribute value msg -> Config msg -> Config msg
+updateConfig attribute config =
+    case attribute of
+        InputStyleAttribute theme ->
+            { config | inputStyle = theme }
+
+        ErrorAttribute isInError ->
+            { config | isInError = isInError }
+
+        HideLabelAttribute hideLabel ->
+            { config | hideLabel = hideLabel }
+
+        PlaceholderAttribute text_ ->
+            { config | placeholder = Just text_ }
+
+        OnBlurAttribute msg ->
+            { config | onBlur = Just msg }
+
+        AutofocusAttribute autofocus_ ->
+            { config | autofocus = autofocus_ }
+
+
+{-| Render the TextInput as HTML.
+The input's label, InputType, and current value are all required. Other attributes are all optional.
+-}
+view : String -> InputType value msg -> List (Attribute value msg) -> value -> Html msg
+view label inputType attributes currentValue =
+    let
+        config =
+            List.foldl updateConfig emptyConfig attributes
+    in
+    view_ label inputType config currentValue
+
+
+{-| Uses the "Writing" input style. See [`Nri.Ui.InputStyles.V2.Theme`](Nri-Ui-InputStyles-V2#Theme).
+-}
+writing : Attribute value msg
+writing =
+    InputStyleAttribute InputStyles.Writing
+
+
+view_ : String -> InputType value msg -> Config msg -> value -> Html msg
+view_ label (InputType inputType) config currentValue =
+    let
+        idValue =
+            generateId label
+
+        placeholder_ =
+            config.placeholder
+                |> Maybe.withDefault label
+
+        maybeStep =
+            if inputType.fieldType == "number" then
+                [ step "any" ]
+
+            else
+                []
+
+        maybeAttr attr maybeValue =
+            maybeValue
+                |> Maybe.map attr
+                |> Maybe.withDefault Extra.none
+    in
+    div
+        [ Attributes.css [ position relative ]
+        ]
+        [ input
+            (maybeStep
+                ++ [ Attributes.id idValue
+                   , css
+                        [ InputStyles.input config.inputStyle config.isInError
+                        , if config.inputStyle == InputStyles.Writing then
+                            Css.Global.withClass "override-sass-styles"
```

Comment:
not sure... all I did in this function was path up how to get the config values from their new places.  I didn't want to accidentally break the Writing TextInput.  But I can try to investigate if this can be removed if you want me to.

Reply:
Hmm it might be a project to figure out, given that we added the override-sass-styles class to get out of actually fixing our inherited styles issues. 🤔 Feel free to investigate if you have the time & energy, but it's not a blocker.

Reply:
okay, I'll leave this alone, then, since we want to get this merged to monolith tomorrow